### PR TITLE
[CARBONDATA-970] getSplits() from carbonScanRDD not ignoring the blocks which are cleared using clean files DDL

### PR DESCRIPTION
**Pull request by sraghunandan**
problem: After clean files DDL is run, the blocks which are deleted are also getting in getSplits() of CarbonScanRDD.scala

Solution: when the segments index map is created in carbonInputFormat.scala , get only the valid tasks which contain the valid blocks

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1745/

**Comment by akashrn5** [comment]
http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1750/   CI build passed fro spark 1.5

File: hadoop/src/main/java/org/apache/carbondata/hadoop/CarbonInputFormat.java [hunk 1]
```diff
@@ -519,6 +518,14 @@ private boolean isValidBlockBasedOnUpdateDetails(
             cacheClient.getSegmentAccessClient().get(tableSegmentUniqueIdentifier);
         segmentIndexMap = segmentTaskIndexWrapper.getTaskIdToTableSegmentMap();
       }
+      Map<SegmentTaskIndexStore.TaskBucketHolder, AbstractIndex> finalMap =
```
Comment (ravipesala) [inline hadoop/src/main/java/org/apache/carbondata/hadoop/CarbonInputFormat.java]: Move the finalMap inside if condition.

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6.2, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/1757/

**Comment by ravipesala** [comment]
LGTM

**Pull request merged**